src/operators/operators.cpp

```diff
@@ -0,0 +1,1445 @@
+#include <deal.II/base/geometry_info.h>
+
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/fe/fe_dgp.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/mapping_fe_field.h>
+#include <deal.II/fe/mapping_q1_eulerian.h>
+
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/hp/q_collection.h>
+#include <deal.II/hp/mapping_collection.h>
+#include <deal.II/hp/fe_values.h>
+
+#include <deal.II/lac/vector.h>
+#include <deal.II/lac/sparsity_pattern.h>
+#include <deal.II/lac/trilinos_sparse_matrix.h>
+#include <deal.II/lac/trilinos_vector.h>
+
+#include <Epetra_RowMatrixTransposer.h>
+#include <AztecOO.h>
+
+#include "ADTypes.hpp"
+#include <Sacado.hpp>
+#include <CoDiPack/include/codi.hpp>
+
+#include "operators.h"
+
+namespace PHiLiP {
+namespace OPERATOR {
+
+//Constructor
+template <int dim, int nstate, typename real>
+OperatorBase<dim,nstate,real>::OperatorBase(
+    const Parameters::AllParameters *const parameters_input,
+    const unsigned int degree,
+    const unsigned int max_degree_input,
+    const unsigned int grid_degree_input)
+    : OperatorBase<dim,nstate,real>(parameters_input, degree, max_degree_input, grid_degree_input, this->create_collection_tuple(max_degree_input, parameters_input))
+{ }
+
+template <int dim, int nstate, typename real>
+OperatorBase<dim,nstate,real>::OperatorBase(
+    const Parameters::AllParameters *const parameters_input,
+    const unsigned int /*degree*/,
+    const unsigned int max_degree_input,
+    const unsigned int grid_degree_input,
+    const MassiveCollectionTuple collection_tuple)
+    : all_parameters(parameters_input)
+    , max_degree(max_degree_input)
+    , max_grid_degree(grid_degree_input)
+    , fe_collection_basis(std::get<0>(collection_tuple))
+    , volume_quadrature_collection(std::get<1>(collection_tuple))
+    , face_quadrature_collection(std::get<2>(collection_tuple))
+    , oned_quadrature_collection(std::get<3>(collection_tuple))
+    , fe_collection_flux_basis(std::get<4>(collection_tuple))
+    , mpi_communicator(MPI_COMM_WORLD)
+    , pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(mpi_communicator)==0)
+{ 
+    pcout<<" Constructing Operators ..."<<std::endl;
+    //setup volume operators
+    allocate_volume_operators();
+    create_vol_basis_operators();
+    build_Mass_Matrix_operators();
+    build_Stiffness_Matrix_operators ();
+    get_higher_derivatives();
+    build_K_operators();
+    get_vol_projection_operators();
+
+    //setup surface operators
+    allocate_surface_operators();
+    create_surface_basis_operators();
+    get_surface_lifting_operators ();
+
+    //setup metric operators
+    allocate_metric_operators();
+    create_metric_basis_operators();
+
+}
+// Destructor
+template <int dim, int nstate, typename real>
+OperatorBase<dim,nstate,real>::~OperatorBase ()
+{
+    pcout << "Destructing Operators..." << std::endl;
+}
+template <int dim, int nstate, typename real>
```

Comment:
I'm guessing this is copy pasted from the DG constructor. Consider moving that function our of DG, and use the same function for both DG and the Operator constructors. Right now, since this is copy pasted, we would have to change code at 2 locations if we find a bug or want to change something in the framework.

Reply:
Yes, this is part of the focus for merge part 2. Since in part 2 I will be changing a fair bit in DG, I figured I will leave it untouched for the first part and just introduce the operators class.